server.cpp

```diff
@@ -85,23 +146,33 @@ int main(int argc, char* argv[])
         tcp::resolver resolver(io_service);
         tcp::acceptor acceptor(io_service, *resolver.resolve(tcp::resolver::query(host, port)));
 
+        namespace pls = std::placeholders;
+
+        char buff[1024];
+
         for (;;)
         {
             tcp::socket socket(io_service);
             acceptor.accept(socket);
 
-            const std::string message = make_daytime_string();
+            const size_t bytes = read(socket, boost::asio::buffer(buff), std::bind(read_complete, buff, pls::_1, pls::_2));
 
-            boost::system::error_code ignored_error;
-            boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+            const std::string msg(buff, bytes);
+            const size_t str_end_pos = msg.find("\r");
+            const std::string start_str = msg.substr(0, str_end_pos);
```

Comment:
This variable is only used once, you can pass it directly to the `Request` constructor.